docs/Makefile

```diff
@@ -0,0 +1,20 @@
+# Minimal makefile for Sphinx documentation
+#
+
+# You can set these variables from the command line, and also
+# from the environment for the first two.
+SPHINXOPTS    ?=
```

Comment:
We don't have to do it in this PR, but I would recommend that in the follow-up where we enable ruff docstring linting we also add `-n -v -W --keep-going`. The `-n` and `-W` make the doc build throw a lot more errors on things like missing docstrings or incorrect crosslinks caused by typos etc, while the `--keep-going` lets you see all the errors rather than stopping on the first one (`-v` is verbose as usual). I'd recommend trying it in this PR and seeing how extensive the errors are, and if there are a lot we can defer to a future PR. It seems like you already added a bunch of missing docstrings, which I imagine was to silence these kinds of warnings.

Reply:
Good idea!